TestDataAccess/JSONReader.cs

```diff
@@ -9,72 +9,63 @@ namespace TestDataAccess
 {
     public class JSONReader
     {
-        public static Dictionary<string, string[]> TestValues;
-        public static Dictionary<string, string[]> TestJsonArrayValues;
-        public static Dictionary<string, string> TestCaseValues;
-        public static int ListCount;
+        private JSONFile JsonFile;
 
-        public JSONReader(JSONFile jsonFile,
-            string testDataKey, int testDataIndex)
+        public JSONReader(JSONFile jsonFile)
+        {
+            this.JsonFile = jsonFile;
+        }
+
+        /// <summary>
+        /// Returns a root level JSON collection value by specifying its key and collection index level.
+        /// </summary>
+        /// <param name="testDataKey"></param>
+        /// <param name="testDataIndex"></param>
+        /// <returns></returns>
+        public Dictionary<string, string> GetKeyAndIndexValue(string testDataKey, int testDataIndex)
         {
             var testDataAtTestDataIndex =
-                testData(jsonFile, testDataKey, testDataIndex);
+                this.GetKeyAndIndexValueFromJObject(this.JsonFile, testDataKey, testDataIndex);
 
-            TestCaseValues = JsonConvert
+            return JsonConvert
                 .DeserializeObject<Dictionary<string, string>>(testDataAtTestDataIndex.ToString());
         }
 
         /// <summary>
-        /// Constructor for a JSONReader to read the testCaseValues from a JSON File
+        /// Returns a nested level JSON collection value by specifying its key, collection index level and parent key and index.
         /// </summary>
         /// <param name="testDataKey"></param>
-        /// <param name="testDataIndex">Index of the test case in JSON file</param>
-        /// <param name="testDataIsInArray">To check if the test data is in an array</param>
-        public JSONReader(JSONFile jsonFile,
-            string testDataKey, int testDataIndex, bool testDataIsInArray)
+        /// <param name="testDataIndex"></param>
+        /// <param name="subKey"></param>
+        /// <param name="subIndex"></param>
+        /// <returns></returns>
+        public Dictionary<string, string> GetNestedKeyAndIndexValue(string testDataKey, int testDataIndex, string subKey, int subIndex)
         {
-            JObject testDataAtTestDataIndex =
-                (JObject)testData(jsonFile, testDataKey, testDataIndex);
-
-            if (!testDataIsInArray)
```

Comment:
Passing bools as parameter values are generally a bad idea. The simplified way to do this is to create a function that can only do one thing... get test data that IS in an array.
Moved into GetArrayThruKeyAndIndexValue

Reply:
The reason its generally a bad idea is because a bool increases the number of permutations for function invocation, which we have to account for in maintenance, and testing. i.e it works when the bool is true, but what if its false? What if thee first parameter is null? etc

-- Also, I know you didnt add this,--- it was already here.